pkg/orchestration/controller.go

```diff
@@ -85,65 +85,105 @@ func (c *Controller) Process(ctx *ctrl.ProcessContext) (retriable bool, err erro
+func (c *Controller) process(logger *zap.Logger, state *orch_v1.State) (b *smith_v1.Bundle, external bool, conflict bool, retriable bool, err error) {
+	entanglerContext, err := c.constructEntanglerContext(state)
+	if err != nil {
+		return nil, false, false, false, err
+	}
+
+	bundle, external, retriable, err := c.entangle(state, entanglerContext)
+	if err != nil {
+		return nil, external, false, retriable, err
+	}
+
+	conflict, retriable, bundle, err = c.createOrUpdateBundle(logger, state, bundle)
+	return bundle, false, conflict, retriable, err
+
+}
+
+func (c *Controller) constructEntanglerContext(state *orch_v1.State) (*wiring.EntangleContext, error) {
 	// Grab the namespace
 	namespaceObj, exists, err := c.NamespaceInformer.GetIndexer().GetByKey(state.Namespace)
 	if err != nil {
-		return false, false, nil, errors.WithStack(err)
+		return nil, errors.WithStack(err)
 	}
 	if !exists {
-		return false, false, nil, errors.Errorf("missing Namespace %q in informer", state.Namespace)
+		return nil, errors.Errorf("missing Namespace %q in informer", state.Namespace)
 	}
 	namespace := namespaceObj.(*core_v1.Namespace)
 
-	// Grab the ConfigMap
-	if state.Spec.ConfigMapName == "" {
-		return false, false, nil, errors.Errorf("configMapName is not provided in State spec for %q", state.GetName())
-	}
 	key := ByConfigMapNameIndexKey(state.Namespace, state.Spec.ConfigMapName)
 	configMapInterface, exists, err := c.ConfigMapInformer.GetIndexer().GetByKey(key)
 	if err != nil {
-		return false, false, nil, errors.WithStack(err)
+		return nil, errors.WithStack(err)
 	}
 	if !exists {
-		return false, false, nil, errors.Errorf("missing ConfigMap %q (key: %q) in informer", state.Spec.ConfigMapName, key)
+		return nil, errors.Errorf("missing ConfigMap %q (key: %q) in informer", state.Spec.ConfigMapName, key)
 	}
 	serviceProperties, err := parseConfigMap(configMapInterface.(*core_v1.ConfigMap))
 	if err != nil {
-		return false, false, nil, errors.WithStack(err)
+		return nil, errors.WithStack(err)
 	}
 
 	serviceName, err := layers.ServiceNameFromNamespaceLabels(namespace.Labels)
 	if err != nil {
-		return false, false, nil, err
+		return nil, err
 	}
 
 	// Entangle the State
-	entangleContext := &wiring.EntangleContext{
+	return &wiring.EntangleContext{
 		ServiceName:       serviceName,
 		Label:             layers.ServiceLabelFromNamespaceLabels(namespace.Labels),
 		ServiceProperties: *serviceProperties,
+	}, nil
+}
+
+func (c *Controller) entangle(state *orch_v1.State, entangleContext *wiring.EntangleContext) (*smith_v1.Bundle, bool /* external */, bool /* retriable */, error) {
+	result := c.Entangler.Entangle(state, entangleContext)
+	switch r := result.(type) {
+	case *wiring.EntangleResultSuccess:
+		return r.Bundle, false, false, nil
+	case *wiring.EntangleResultFailure:
+		return nil, r.IsExternalError, r.IsRetriableError, errors.Wrapf(r.Error, "failed to wire up Bundle for State %q", state.Name)
```

Comment:
This message will be in the condition on the State object - no need to include it's name in the message

Reply:
Yeah, this was in the old error message. I'll remove it.